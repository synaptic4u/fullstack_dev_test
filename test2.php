<?php
/**
 * QUESTION 2
 *
 * Using the data stored in the database
 * show a list of products with their prices
 * grouped by category.
 * The categories should be listed in alphabetical order.
 * The products within those categories should also be listed in alphabetical order.
 * Products with no category will be categorized as "Uncategorized".
 * If there are no results, then it should just say "There are no results available."
 *
 * Please make sure your code runs as effectively as it can.
 */


?>
<!DOCTYPE html>
<html>
<head>
    <title>Test2</title>
</head>
<body>
<h1>Products</h1>

<?php

// Prepare the SQL query to get products with categories
$query = "
    SELECT 
        IFNULL(c.category, 'Uncategorized') AS category, 
        p.product, 
        p.price 
    FROM 
        products p
    LEFT JOIN 
        categories c ON p.category_id = c.id
    ORDER BY 
        category ASC, 
        p.product ASC
";

// Prepare the statement
if ($stmt = $con->prepare($query)) {
    // Execute the statement
    $stmt->execute();

    // Get the result set
    $result = $stmt->get_result();

    // Check if there are any rows
    if ($result->num_rows > 0) {
        $currentCategory = '';
        while ($row = $result->fetch_assoc()) {
            // Check if we're in a new category
            if ($row['category'] !== $currentCategory) {
                // Close the previous list if necessary and open a new one
                if (!empty($currentCategory)) {
                    echo "</ul>";
                }
                // Print category header
                echo "<h2>" . htmlspecialchars($row['category']) . "</h2>";
                echo "<ul>";
                $currentCategory = $row['category'];
            }
            // Print product name and price
            echo "<li>" . htmlspecialchars($row['product']) . " - $" . number_format($row['price'], 2) . "</li>";
        }
        echo "</ul>"; // Close the last list
    } else {
        echo "<p>There are no results available.</p>";
    }

    // Close the statement
    $stmt->close();
} else {
    echo "Failed to prepare the statement: " . $con->error;
}
$con->close();
?>

</body>
</html>
