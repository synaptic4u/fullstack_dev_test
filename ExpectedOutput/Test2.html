<!DOCTYPE html>
<html>
<head>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
    <title>Test2</title>
</head>
<body cz-shortcut-listen="true">
<h1>Products</h1>

<h2>First Person Shooters</h2>
<table width="400">
    <tbody>
    <tr>
        <th align="left">Product</th>
        <th align="right">Price</th>
    </tr>

    <tr>
        <td>Counter-Strike: Global Offensive</td>
        <td align="right">R 159.00</td>
    </tr>

    <tr>
        <td>DOOM</td>
        <td align="right">R 799.00</td>
    </tr>

    <tr>
        <td>Farcry Primal</td>
        <td align="right">R 549.00</td>
    </tr>

    <tr>
        <td>Left 4 Dead 2</td>
        <td align="right">R 219.00</td>
    </tr>
    </tbody>
</table>
<h2>Real Time Strategy Games</h2>
<table width="400">
    <tbody>
    <tr>
        <th align="left">Product</th>
        <th align="right">Price</th>
    </tr>

    <tr>
        <td>Age of Empires 3: Complete Collection</td>
        <td align="right">R 409.00</td>
    </tr>

    <tr>
        <td>Age of Mythology: Extended Edition</td>
        <td align="right">R 319.00</td>
    </tr>

    <tr>
        <td>Command &amp; Conquer: Red Alert 3</td>
        <td align="right">R 219.00</td>
    </tr>

    <tr>
        <td>Homeworld Remastered Collection</td>
        <td align="right">R 359.00</td>
    </tr>

    <tr>
        <td>Rise of Nations: Extended Edition</td>
        <td align="right">R 219.00</td>
    </tr>
    </tbody>
</table>
<h2>Role Playing Games</h2>
<table width="400">
    <tbody>
    <tr>
        <th align="left">Product</th>
        <th align="right">Price</th>
    </tr>

    <tr>
        <td>Dark Souls 3</td>
        <td align="right">R 799.00</td>
    </tr>

    <tr>
        <td>Fable Anniversary</td>
        <td align="right">R 359.00</td>
    </tr>

    <tr>
        <td>Grim Dawn</td>
        <td align="right">R 269.00</td>
    </tr>

    <tr>
        <td>The Elder scrolls V: Skyrim</td>
        <td align="right">R 219.00</td>
    </tr>

    <tr>
        <td>The Witcher 3: Wild Hunt</td>
        <td align="right">R 399.99</td>
    </tr>
    </tbody>
</table>
<h2>Uncategorized</h2>
<table width="400">
    <tbody>
    <tr>
        <th align="left">Product</th>
        <th align="right">Price</th>
    </tr>

    <tr>
        <td>7 Days to Die</td>
        <td align="right">R 269.00</td>
    </tr>

    <tr>
        <td>DayZ</td>
        <td align="right">R 429.00</td>
    </tr>

    <tr>
        <td>Portal Bundle</td>
        <td align="right">R 246.00</td>
    </tr>

    <tr>
        <td>Tales of Berseria</td>
        <td align="right">R 757.00</td>
    </tr>
    </tbody>
</table>

</body>
</html>